utility/src/test/java/org/oppia/android/util/utility/NetworkConnectionTestUtil.kt

```diff
@@ -0,0 +1,35 @@
+package org.oppia.android.util.utility
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import org.robolectric.Shadows
+import org.robolectric.shadows.ShadowNetworkInfo
+
+/** Test utility to modify [ShadowNetworkInfo] during the test. */
+class NetworkConnectionTestUtil {
+
+  companion object {
+    /**
+     * Sets the [ShadowNetworkInfo] during the test.
+     *
+     * @param context: the context of the dummy network connection.
+     * @param status: the type of network.
+     * @param networkState: [status] of the network connection.
+     */
+    fun setNetworkInfo(context: Context, status: Int, networkState: NetworkInfo.State) {
+      Shadows.shadowOf(
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+      )
```

Comment:
Adding onto my suggestion above to make the util class injectable: this could be a by lazy field to avoid needing to fetch it multiple times.

Note that only the getSystemService result should be stored in a field, not the shadow itself (the Robolectric team strongly discourages caching shadow objects themselves).

Reply:
I'm not able to fully understand what you are trying to say. How can we store the `getSystemService` result only? Are you referring to `ShadowNetworkInfo.newInstance`?

Reply:
Will address other comments related to `NetworkConnectionTestUtil` once this is clarified.

Reply:
I'm suggesting something like this:

```kotlin
class NetworkConnectionTestUtil @Inject constructor(
  @ApplicationContext private val applicationContext: Context
) {
  private val connectivityManager by lazy {
    context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
  }

  ...
```

And then when we want to use it, we just call ``Shadows.shadowOf(connectivityManager)``.

Does this help clarify?

Reply:
Thanks for the clarification @BenHenning. It really helped. I have made the changes as suggested. PTAL. Thanks.